sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -427,6 +427,11 @@ private[hive] class HiveClientImpl(
   }
 
   override def alterTable(tableName: String, table: CatalogTable): Unit = withHiveState {
+    // Not allowed to switch the table type
+    if (table.properties.contains("EXTERNAL")) {
```

Comment:
The reason why I put here is because setting `EXTERNAL` will not change the table type when using `InMemoryCatalog`.

Will move it for unifying both catalog. Thanks!